File: src/transformers/features/AssertTransformer.ts
import { AssertProgrammer } from "../../programmers/AssertProgrammer";

import { ITransformProps } from "../ITransformProps";
import { GenericTransformer } from "../internal/GenericTransformer";

export namespace AssertTransformer {
  export const transform =
    (config: AssertProgrammer.IConfig) => (props: ITransformProps) =>
      GenericTransformer.scalar({
        ...props,
        method: config.equals
          ? config.guard
            ? "assertGuardEquals"
            : "assertEquals"
          : config.guard
            ? "assertGuard"
            : "assert",
        write: (x) =>
          AssertProgrammer.write({
            ...x,
            config,
          }),
      });
}

File: packages/transform/src/features/AssertTransformer.go
package features

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var AssertTransformer = assertTransformerNamespace{}

type assertTransformerNamespace struct{}

func (assertTransformerNamespace) Transform(equals bool, guard bool) internal.Task {
	return internal.GenericTransformer.Scalar(programmers.AssertProgrammer.Write)
}
